qiskit_ibm/runtime/ibm_runtime_service.py

```diff
@@ -107,15 +107,17 @@ def __init__(self, provider: 'ibm_provider.IBMProvider') -> None:
         self._ws_url = provider.credentials.runtime_url.replace('https', 'wss')
         self._programs = {}  # type: Dict
 
-    def pprint_programs(self, refresh: bool = False, detailed: bool = False) -> None:
+    def pprint_programs(self, refresh: bool = False, detailed: bool = False,
+                        name: Optional[str] = "") -> None:
         """Pretty print information about available runtime programs.
 
         Args:
             refresh: If ``True``, re-query the server for the programs. Otherwise
                 return the cached value.
             detailed: If ``True`` print all details about available runtime programs.
+            name: Program name.
```

Comment:
Could you explain how the match is done? Is it a starts with search or a contains search?